pkg/channel/distributed/receiver/producer/producer.go

```diff
@@ -117,6 +120,8 @@ func (p *Producer) ProduceKafkaMessage(ctx context.Context, channelReference eve
 		return err
 	}
 
+	producerMessage.Headers = append(producerMessage.Headers, commonutil.SerializeTrace(trace.FromContext(ctx).SpanContext())...)
+
```

Comment:
Can we add a comment here to make clear what is happening...

Reply:
Absolutely